packages/docs/developer-resources/walkthroughs/web-dapp.md

```diff
@@ -0,0 +1,299 @@
+    const dequeue = await governance.getDequeue()
+
+    const fetchedProposals = await Promise.all(
+      dequeue.map(async (id) => ({ id, ...(await governance.getProposalRecord(id)) }))
+    )
+    setProposals(fetchedProposals)
+  }, [kit])
+
+  useEffect(() => {
+    fetchProposals()
+  }, [fetchProposals])
+
+  return (
+    <table>
+      <thead>
+        <tr>
+          <th>ID</th>
+          <th>Status</th>
+          <th>Description URL</th>
+        </tr>
+      </thead>
+      <tbody>
+        {proposals.map((proposal) => (
+          <tr>
+            <td>{proposal.id.toString()}</td>
+            <td>{proposal.passed ? 'Passed' : proposal.approved ? 'Approved' : 'Not approved'}</td>
+            <td>
+              <a href={proposal.descriptionURL} target="_blank">
+                Description link
+              </a>
+            </td>
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  )
+}
+```
+
+This works pretty well however it makes sense to additionally show whether the user has voted on any given dequeued governance proposal. To show that information, we can amend our `fetchProposals` function as follows
+
+```javascript
+const fetchProposals = useCallback(async () => {
+  if (!address) {
+    return
+  }
+
+  const governance = await kit.contracts.getGovernance()
+  const dequeue = await governance.getDequeue()
+
+  const fetchedProposals = await Promise.all(
+    dequeue.map(async (id) => {
+      const [record, voteRecord] = await Promise.all([
+        governance.getProposalRecord(id),
+        governance.getVoteRecord(address, id),
+      ])
+
+      return {
+        id,
+        ...record,
+        vote: voteRecord ? voteRecord.value : undefined,
+      }
+    })
+  )
+  setProposals(fetchedProposals)
+}, [kit, address])
+```
+
+Now we have access to whether the user voted on this proposal, we can render that information in our table.
+
+```javascript
+return (
+  <table>
+    <thead>
+      <tr>
+        <th>ID</th>
+        <th>Status</th>
+        <th>Description URL</th>
+        <th>Voted</th>
+      </tr>
+    </thead>
+    <tbody>
+      {proposals.map((proposal) => (
+        <tr>
+          <td>{proposal.id.toString()}</td>
+          <td>{proposal.passed ? 'Passed' : proposal.approved ? 'Approved' : 'Not approved'}</td>
+          <td>
+            <a href={proposal.descriptionURL} target="_blank">
+              Description link
+            </a>
+          </td>
+          <td>{proposal.vote ?? 'No vote yet'}</td>
+        </tr>
+      ))}
+    </tbody>
+  </table>
+)
+```
+
+### Locking Celo (optional)
+
+A prerequisite to [voting on Celo governance proposals](celo-owner-guide/voting-governance.md) is having locked Celo to vote with. We won't cover the various flows for locking, unlocking and relocking Celo in this tutorial but you can check the implementation in [Celo Tools](https://github.com/alexbharley/celo-tools) or take inspiration from the following script:
```

Comment:
Linter doesn't like this path to the file
from CI:
`194:19-194:95  warning  Link to unknown file: `celo-owner-guide/voting-governance.md`. Did you mean `../../celo-holder-guide/voting-governance.md`  missing-file  remark-validate-links`